controllers/imagerepository_controller.go

```diff
@@ -432,3 +442,46 @@ func (r *ImageRepositoryReconciler) recordSuspension(ctx context.Context, imageR
 		r.MetricsRecorder.RecordSuspend(*objRef, imageRepo.Spec.Suspend)
 	}
 }
+
+func parseAuthMap(config dockerConfig) (map[string]authn.AuthConfig, error) {
+	auth := map[string]authn.AuthConfig{}
+	for url, entry := range config.Auths {
+		host, err := getURLHost(url)
+		if err != nil {
+			return nil, err
+		}
+
+		auth[host] = entry
+	}
+
+	return auth, nil
+}
+
+func getURLHost(urlStr string) (string, error) {
+	if urlStr == "http://" || urlStr == "https://" {
+		return "", errors.New("Empty url")
+	}
+
+	// ensure url has https:// or http:// prefix
+	// url.Parse won't parse the ip:port format very well without the prefix.
+	if !strings.HasPrefix(urlStr, "http://") && !strings.HasPrefix(urlStr, "https://") {
+		urlStr = fmt.Sprintf("https://%s/", urlStr)
+	}
+
+	// Some users were passing in credentials in the form of
+	// http://docker.io and http://docker.io/v1/, etc.
+	// So strip everything down to the host.
+	// Also, the registry might be local and on a different port.
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return "", err
+	}
+
+	if u.Host == "" {
+		return "", errors.New(fmt.Sprintf(
+			"Invalid registry auth url: %s. Must be a valid http url (e.g. https://gcr.io/v1/) or prepending with 'https' makes it a valid http url",
```

Comment:
This one is tricky to word isn't it? I'm not happy with my earlier suggestion, now :-S 

How about this:

```suggestion
			"Invalid registry auth key: %s. Expected an HTTPS URL (e.g. 'https://gcr.io/v1/' or 'https://gcr.io'), or the same without the 'https://' (e.g., 'gcr.io/v1/' or 'gcr.io')"
```

Reply:
Blah, I'm just bikeshedding now -- your one is :+1:

Reply:
Let's replace GCR with docker.io please, this would look very strange to someone using AWS or Azure.

Reply:
(incorporated that into the click-to-commit)